### ISSUE TITLE
Snake case, owned entities with inheritance

### ISSUE THREAD

**Issue by tuggernuts** [efcore/EFCore.NamingConventions#87]
Hi. This is perhaps somewhat related to [Issue #69](https://github.com/efcore/EFCore.NamingConventions/issues/69#issue-815653536)? Or, maybe closer to [this closed one](https://github.com/efcore/EFCore.NamingConventions/issues/62#issue-791318721)?

In this case, migrations fail in a Table-per-Type scenario, where the base class has an owned entity.

```csharp
    public class DataContext : DbContext
    {
        public DbSet<Person> People { get; set; }
        public DbSet<User> Users { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer("Data Source=(localdb)\\ProjectsV13; InitialCatalog=EFCoreTests").UseSnakeCaseNamingConvention();
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<User>().ToTable("users"); // required for TPT
            modelBuilder.Entity<Person>().OwnsOne(p => p.Address);
            base.OnModelCreating(modelBuilder);
        }
    }
```

```csharp
public class Person
    {
        public int Id { get; private set; }
        public string FirstName { get; private set; }
        public string LastName { get; private set; }
        public Address Address { get; private set; }
    }

    public class Address
    {
        public string Street { get; private set; }
        public string City { get; private set; }
        public string State { get; private set; }
        public string Zip { get; private set; }
    }

   public class User : Person
    {
        public string Role { get; private set; }

    }
```

In this case, the migration will fail with:
```
System.InvalidOperationException: Cannot use table 'people' for entity type 'Person' since it is being used for entity type 'Address' and the name 'PK_people' of the primary key {'Id'} does not match the name 'pk_people' of the primary key {'PersonId'}.
```
By removing `UseSnakeCaseNamingConvention()`, the migration will succeed.

I'm using version 5.0.2 with EF Core 5.0.7.

**Comment by Azmond** [comment]
I have the same problem in version 5.0.2. 
I tried to build the library of the current source. Migrations were successful 😃

**Comment by manelferreira** [comment]
I'm also having this problem.

One observation: If the CurrentStatus property (mapped with OwnsOne) is removed from the class and only the History property (mapped with OwnsMany) is left, the migration is generated successfully.

```
public class LoanContract : Entity<int>
{
      public LoanContractStatus CurrentStatus { get; set; }
      private List<LoanContractHistory> _history = new List<LoanContractHistory>();
      public List<LoanContractHistory> History { get { return _history; } }
}

public class SomeLenderLoanContract : LoanContract
{
        public string LenderContractId { get; set; }
}
```

The ef core entity configuration:
```
public class LoanContractConfiguration : IEntityTypeConfiguration<LoanContract>
{
	public void Configure(EntityTypeBuilder<LoanContract> builder)
	{
            builder
                .ToTable("loan_contract");

            builder
                .HasKey(x => x.Id);

            builder
                .OwnsOne(x => x.CurrentStatus, status => 
                {   
                    status.Property(x => x.Status)
                        .HasConversion<int>();
                });

            builder.OwnsMany(x => x.History, history => 
            {
                history.OwnsOne(x => x.Status, status => 
                {
                    status.Property(p => p.Status)
                        .IsRequired()
                        .HasColumnName("status");

                    status.HasIndex(p => p.Status);
                });
	}
}
```

When I run the dotnet command to generate a new migration, the exception is thrown:

```
System.InvalidOperationException: Cannot use table 'loan_contract' for entity type 'LoanContract' since it is being used for entity type 'LoanContract.CurrentStatus#LoanContractStatus' and the name 'PK_loan_contract' of the primary key {'Id'} does not match the name 'pk_loan_contract' of the primary key {'LoanContractId'}.
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.ValidateSharedTableCompatibility(IReadOnlyList`1 mappedTypes, String tableName, String schema, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.ValidateSharedTableCompatibility(IModel model, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.Validate(IModel model, IDiagnosticsLogger`1 logger)
   at Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure.NpgsqlModelValidator.Validate(IModel model, IDiagnosticsLogger`1 logger)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.ValidatingConvention.ProcessModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnModelFinalized(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Internal.Model.FinalizeModel()
   at Microsoft.EntityFrameworkCore.ModelBuilder.FinalizeModel()
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, ModelDependencies modelDependencies)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.GetModel(DbContext context, IConventionSetBuilder conventionSetBuilder, ModelDependencies modelDependencies)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServicesBuilder.<>c.<TryAddCoreServices>b__7_3(IServiceProvider p)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.DbContext.Microsoft.EntityFrameworkCore.Infrastructure.IInfrastructure<System.IServiceProvider>.get_Instance()
   at Microsoft.EntityFrameworkCore.Infrastructure.Internal.InfrastructureExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Cannot use table 'loan_contract' for entity type 'LoanContract' since it is being used for entity type 'LoanContract.CurrentStatus#LoanContractStatus' and the name 'PK_loan_contract' of the primary key {'Id'} does not match the name 'pk_loan_contract' of the primary key {'LoanContractId'}.
```

The final error message:
Cannot use table 'loan_contract' for entity type 'LoanContract' since it is being used for entity type 'LoanContract.CurrentStatus#LoanContractStatus' and the name 'PK_loan_contract' of the primary key {'Id'} does not match the name 'pk_loan_contract' of the primary key {'LoanContractId'}.

**Comment by wagnergraciano** [comment]
I'm also having the same problem.

**Comment by roji** [comment]
I've recently fixed #112 and #69 for 6.0.0 - I can't see the problems above after that. If you still run into trouble after upgrading to 6.0.0, please open a new issue with a full, minimal model and I'll take a look.

**Comment by tuggernuts** [comment]
> I've recently fixed #112 and #69 for 6.0.0 - I can't see the problems above after that. If you still run into trouble after upgrading to 6.0.0, please open a new issue with a full, minimal model and I'll take a look.

Yes, > 6.0.0 has been working. Thank you for your work.

**Comment by roji** [comment]
Great, thanks.
